pylint/config/config_initialization.py

```diff
@@ -81,8 +81,9 @@ def _config_initialization(
             unrecognized_options.append(opt[1:])
     if unrecognized_options:
         msg = ", ".join(unrecognized_options)
+        linter._arg_parser.print_usage()
```

Comment:
```suggestion
        linter._arg_parser.error("Unrecognized option found: Q")
```
What's the result if we do that ?

Reply:
Hmm for example:
```
import argparse
parser = argparse.ArgumentParser()
parser.add_argument('--foo')
parser.add_argument('cmd')
parser.add_argument('rest', nargs='*', type=int)
parser.parse_known_args('doit 1 --foo bar 2 3'.split())
parser.error("My error")
```
Does:
```
usage: -c [-h] [--foo FOO] cmd [rest [rest ...]]
-c: error: My error
```
So it seems we do not have the automated usage message anymore ?

Reply:
That's has been the case for quite some time now, and probably a different issue. I think the issue here is that we're kind of duplicating the message by doing both `add_message` and then `error()`. We probably need to choose one. For configuration files we also use `add_message` but then continue instead of exiting. That's why I think doing `add_message` and then existing with `sys.exit(1)` would be best as that indicates a fatal warning in our system.

Reply:
A parse error made sense for command line options, an add_message made sense for a problem in the configuration, now we're handling both at the same time it seems. We need to choose if we want to exit for bad configurations, I think it's counter intuitive to have an add_message for bad option in CLI, it's supposed to be parsed before we enter the program.

Reply:
```console
❯ pylint --Q=1
usage: pylint [options]
pylint: error: Unrecognized option found: Q=1
```